src/node_api.cc

```diff
@@ -827,6 +827,339 @@ napi_status ConcludeDeferred(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+class TsFn: public node::AsyncResource {
```

Comment:
style nits: I'd very much prefer to spell out `ThreadSafeFunction`, and we use spaces before the `:` in a class declaration

If you need abbreviations for individual functions, I think an `using` line might do the trick